File: frontend/mobile_app/lib/models/session.dart
/// The signed-in sales associate.
class Salesperson {
  const Salesperson({
    required this.id,
    required this.name,
    this.title,
    this.username,
  });

  factory Salesperson.fromJson(Map<String, dynamic> json) => Salesperson(
    id: json['id'] as String,
    name: json['name'] as String,
    title: json['title'] as String?,
    username: json['username'] as String?,
  );

  final String id;
  final String name;
  final String? title;

  /// Login handle returned by the auth API; used to pre-fill the profile.
  final String? username;

  Map<String, dynamic> toJson() => <String, dynamic>{
    'id': id,
    'name': name,
    'title': title,
    if (username != null) 'username': username,
  };
}

/// Parsed content of a pairing QR code:
/// `http://<box-ip>:<port>/pair?token=<pairingToken>`.
class PairingInfo {
  const PairingInfo({
    required this.host,
    required this.port,
    required this.token,
  });

  /// Parses a pairing URL. Returns null if it isn't a valid pairing link.
  static PairingInfo? tryParse(String raw) {
    final Uri? uri = Uri.tryParse(raw.trim());
    if (uri == null) return null;
    final String? token = uri.queryParameters['token'];
    if (token == null || token.isEmpty) return null;
    return PairingInfo(
      host: uri.host,
      port: uri.hasPort ? uri.port : 8080,
      token: token,
    );
  }

  final String host;
  final int port;
  final String token;

  Uri get wsUri => Uri(scheme: 'ws', host: host, port: port, path: '/ws');
}

/// An active showroom session binding a salesperson to a display.
class SessionInfo {
  const SessionInfo({
    required this.sessionId,
    required this.displayId,
    required this.salesperson,
  });

  final String sessionId;
  final String displayId;
  final Salesperson salesperson;
}

File: frontend-rn/mobile_app/src/models/session.ts
/// The signed-in sales associate. Ported from the Flutter mobile `Salesperson`.
export class Salesperson {
  readonly id: string;
  readonly name: string;
  readonly title?: string;
  /// Login handle returned by the auth API; used to pre-fill the profile.
  readonly username?: string;

  constructor(init: { id: string; name: string; title?: string; username?: string }) {
    this.id = init.id;
    this.name = init.name;
    this.title = init.title;
    this.username = init.username;
  }

  static fromJson(json: any): Salesperson {
    return new Salesperson({
      id: json.id as string,
      name: json.name as string,
      title: json.title as string | undefined,
      username: json.username as string | undefined,
    });
  }

  toJson(): Record<string, any> {
    const out: Record<string, any> = { id: this.id, name: this.name, title: this.title ?? null };
    if (this.username != null) out.username = this.username;
    return out;
  }
}

/// Parsed content of a pairing QR code:
/// `http://<box-ip>:<port>/pair?token=<pairingToken>`.
export class PairingInfo {
  readonly host: string;
  readonly port: number;
  readonly token: string;

  constructor(init: { host: string; port: number; token: string }) {
    this.host = init.host;
    this.port = init.port;
    this.token = init.token;
  }

  static tryParse(raw: string): PairingInfo | null {
    let uri: URL;
    try {
      uri = new URL(raw.trim());
    } catch {
      return null;
    }
    const token = uri.searchParams.get('token');
    if (token == null || token.length === 0) return null;
    const port = uri.port ? parseInt(uri.port, 10) : 8080;
    return new PairingInfo({ host: uri.hostname, port, token });
  }

  get wsUrl(): string {
    return `ws://${this.host}:${this.port}/ws`;
  }
}

/// An active showroom session binding a salesperson to a display.
export class SessionInfo {
  readonly sessionId: string;
  readonly displayId: string;
  readonly salesperson: Salesperson;

  constructor(init: { sessionId: string; displayId: string; salesperson: Salesperson }) {
    this.sessionId = init.sessionId;
    this.displayId = init.displayId;
    this.salesperson = init.salesperson;
  }
}
